recipes/repo.rb

```diff
@@ -374,415 +374,63 @@
   end
 end
 
+
 # Pulp setup
-pulp_data_directory = '/var/repos/.pulp'
-user 'pulp' do
-  manage_home true
-  system true
-  shell '/usr/sbin/nologin'
-  comment 'Pulp content manager'
-  uid '1200'
-end
-group 'docker' do
-  append true
-  members ['pulp']
-  action [:manage]
-end
-group 'gpg-vault' do
-  append true
-  members ['pulp']
-  action [:manage]
+node['ros_buildfarm']['rpm_repos'].keys.each do |dist|
+  dist_dir = "/var/repos/#{dist}_pulp"
+  directory dist_dir do
+    action :delete
+    recursive true
+  end
 end
 
-execute 'gpg-init' do
-  command 'gpg -K'
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  creates '/home/pulp/.gnupg'
-end
-cookbook_file '/home/pulp/.gnupg/gpg.conf' do
-  source 'gpg-vault.conf'
-  owner 'pulp'
-  group 'pulp'
-  mode '0600'
+0.upto(10) do |i|
+  service "pulp-worker@#{i}" do
+    action [:stop, :disable]
+    only_if { File.exist?("/etc/systemd/system/pulp-worker@.service") }
+  end
 end
-execute "gpg --import /var/repos/repos.key" do
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  not_if "gpg --list-keys #{gpg_key['fingerprint']}"
+file "/etc/systemd/system/pulp-worker@.service" do
+  action :delete
 end
-execute 'trust_public_key' do
-  command "(echo 5; echo y; echo save) | gpg --command-fd 0 --no-tty --no-greeting -q --edit-key #{gpg_key['fingerprint']} trust"
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  not_if "gpg --list-keys #{gpg_key['fingerprint']} | grep -q '\\[ultimate\\]'"
+%w(pulp-api-endpoint pulp-content-endpoint pulp-resource-manager pulp-rsync-endpoint).each do |service_name|
+  service service_name do
+    action [:stop, :disable]
+    only_if { File.exist?("/etc/systemd/system/#{service_name}.service") }
+  end
+  file "/etc/systemd/system/#{service_name}.service" do
+    action :delete
+  end
 end
 
-directory pulp_data_directory do
-  owner 'pulp'
-  mode '0700'
-end
-directory "#{pulp_data_directory}/media" do
-  owner 'pulp'
-end
-directory "#{pulp_data_directory}/docker_image" do
-  owner 'pulp'
+file '/usr/local/bin/systemd-docker' do
+  action :delete
 end
 
-cookbook_file "#{pulp_data_directory}/initialize.py" do
-  owner 'pulp'
-  source 'pulp/rpm_repo_init.py'
-  mode '0600'
+service 'postgresql' do
+  action [:stop, :disable]
+  only_if { File.exist?('/usr/lib/systemd/system/postgresql.service') }
 end
 
-package 'redis'
```

Comment:
I was concerned that it might be possible that redis or postgres might be installed for other reasons, and I didn't want to break anything.

If you're confident that they're not otherwise needed, we can remove the packages.

Reply:
> If you're confident that they're not otherwise needed, we can remove the packages.

Do not know the deployment of the ROS repo good enough to say that nothing else is using redis. Fine to go with the safest approach.